docs/DeveloperGuide.md

```diff
@@ -1,34 +1,670 @@
+
+### 3.1 Architecture
+
+This section elaborates on the high-level architecture of the **LifEasier** application. It provides a brief introduction to each component, and how these components interact with one another. Figure 3.1-1 provides an overview of the architecture.
+
+![Figure 3.1-1](images/DeveloperGuide/Figure%203.1-1.png) 
+_Figure 3.1-1 Architecture diagram for LifEasier_
+
+LifEasier is comprised of 7 components, which are listed below together with their functions:
+
+1. `LifEasier`: The main class of the application.
+1. `Ui`: Displays messages to the user, and takes in the user’s commands
+1. `Parser`: Understands the user’s commands, and creates the necessary Command objects.
+1. `Command`: Carries out the user’s command.
+1. `Model`: Holds the user's task and note data in memory.
+1. `Storage`: Reads and writes data from the hard disk.
+
+Each of these components are expanded on in more detail in their respective sections.
+
+Figure 3.1-2 illustrates the Sequence diagram for how each class interacts with one another through an example  “addDeadline” command.
+
+![Figure 3.1-2](images/DeveloperGuide/Figure 3.1-2.png)  
+_Figure 3.1-2. Sequence diagram of “addDeadline”._
+
+### 3.2 Components
+
+This section expands on the various components that were first seen in the Architecture section.
+
+
+#### 3.2.1 LifEasier Component
+
+The **LifEasier** component is the main class of the application. It initialises all other components,
+ and is used as the centre of all other components.
+
+
+#### 3.2.2 UI Component
+
+The `UI` component reads the uthe user. It consists ofser’s inputs and displays messages and content to  a main `Ui`,
+ a `ScheduleUi` and a `TimetableUi`. The `ScheduleUi` handles all outputs to do with displaying a user’s schedule.
+  To do this, it uses the `TimetableUi` to display a specialised timetable view of the weekly schedule. 
+  The main `Ui` handles everything else, such as displaying general messages and reading the user’s inputs. 
+
+
+#### 3.2.3 Parser Component
+
+The `Parser` component takes the user’s inputs from the `Ui` component, and makes sense of these commands. If the
+ commands are incomplete, it calls the `Ui` component to prompt the user for more input until the commands have 
+ the parameters required to execute. Once this condition is fulfilled, it returns a `Command` object to **LifEasier**
+  for the commands to be executed.
+  
+  
+#### 3.2.4 Command Component
+
+The `Command` component consists of an abstract `Command` class and the many different commands that inherit the 
+`Command` class. What is important to note is that each type of command class (e.g. `addLessonCommand`, `showNotesCommand`)
+ implements an abstract `execute()` method that carries out the command. Figure 3.2.4-1 below shows the classes that 
+ inherit from the `Command` class.
+ 
+ ![Figure 3.2.4-1](images/DeveloperGuide/Figure%203.2.4-1.png)  
+ _Figure 3.2.4-1: The classes that inherit from the Command class_
+ 
```

Comment:
![image](https://user-images.githubusercontent.com/60388723/97528406-a79b1e00-19e8-11eb-9628-d116f6315335.png)
I feel that the class diagram is very clear in showing how the command classes are structured and how each command overrides the abstract method